niworkflows/interfaces/images.py

```diff
@@ -670,25 +670,38 @@ class SignalExtraction(SimpleInterface):
     output_spec = SignalExtractionOutputSpec
 
     def _run_interface(self, runtime):
+        img = nb.load(self.inputs.in_file)
         mask_imgs = [nb.load(fname) for fname in self.inputs.label_files]
-        if len(mask_imgs) == 1:
+        multi_index_mask = False
+        if len(mask_imgs) == 1 and len(mask_imgs[0].shape) == 4:
             mask_imgs = nb.four_to_three(mask_imgs[0])
 
-        masks = [mask_img.get_data().astype(np.bool) for mask_img in mask_imgs]
+        for i, mask in enumerate(mask_imgs):
+            if img.shape != mask.shape:
+                mask_imgs[i] = nli.resample_to_img(
```

Comment:
I would enforce that masks and image have the same shape and zooms, raise a RuntimeError otherwise. Let the responsibility of resampling be outside this interface.

Reply:
That permits the image to be loaded after the masks, as it was before.

Reply:
NotImplementedError gets raised if mask and img are of different sizes.